test/html.test.js

```diff
@@ -89,6 +89,117 @@ describe('html', function() {
       html.patch(elem, jml);
       assert.equal(elem.outerHTML, '<div id="a1"></div>');
     });
+
+    it('should update an element given JsonML with raw (Markup) instance', function() {
+      const rawStr = new html.raw('foo');
+      assert.equal(rawStr.value, 'foo');
+      assert.equal(rawStr.toString(), 'foo');
+      const elem = document.createElement('div');
+      const jml = ['', rawStr];
+      html.patch(elem, jml);
+      assert.equal(elem.outerHTML, '<div>foo</div>');
+    });
+
+    describe('element attributes', function() {
+      it('should be able to update element attributes', function() {
+        const elem = document.createElement('input');
+        const attrs = {
+          id: 'my-el-1',
+          name: 'my-best-el',
+          className: 'foo bar baz',
+          disabled: true,
+        };
+        const jml = ['', attrs];
+        html.patch(elem, jml);
+        assert.equal(elem.outerHTML, `<input id="${attrs.id}" name="${attrs.name}" class="${attrs.className}" disabled="">`);
+      });
+
+      it('should be able to update element styles through attributes', function() {
+        const elem = document.createElement('input');
+        // NOTE: JsonML style attribute must provide the literal inline CSS string value.
+        // An object of DOM style properties (e.g. marginTop) will not be applied as expected.
+        const attrs = {
+          style: 'color: blue; margin-top: 25px;',
+        };
+        const jml = ['', attrs];
+        html.patch(elem, jml);
+        assert.equal(elem.outerHTML, `<input style="${attrs.style}">`);
+      });
+
+      it('should be able to update set event handlers through attributes', function(done) {
+        const elem = document.createElement('input');
+        // NOTE: event names must be lowercase. camelCase will currently fail
+        // click to complete the test otherwise the test will fail due to timeout
+        html.patch(elem, ['', { onclick: () => done() }]);
+        elem.dispatchEvent(new document.defaultView.Event('click'));
```

Comment:
Another future patch perhaps. `onClick` will not work. A simple `toLowerCase()` would probably do a world of good.